packages/core/src/server/open.ts

```diff
@@ -107,11 +107,15 @@ export function resolveUrl(str: string, base: string): string {
 
 const normalizeOpenConfig = (
   config: NormalizedConfig,
-): { targets: string[]; before?: () => Promise<void> | void } => {
+): {
+  targets: string[];
+  before?: () => Promise<void> | void;
+  skip?: boolean;
+} => {
   const { open } = config.server;
 
   if (typeof open === 'boolean') {
-    return { targets: [] };
+    return { targets: [], skip: !open };
```

Comment:
This doesn't seem necessary, the server plugin already checks if `open` is false:

https://github.com/web-infra-dev/rsbuild/blob/main/packages/core/src/plugins/server.ts#L15

Reply:
Then there must be something else wrong as 

```
import { defineConfig } from '@rsbuild/core';
import { pluginReact } from '@rsbuild/plugin-react';

export default defineConfig({
  plugins: [pluginReact()],
  server: {
    open: false,
  },
});
```
still opens the browser after the server started.

Reply:
Are you using the `--open` CLI flag? The CLI flag takes precedence over the configurations,

Reply:
oh no, it is part of the package dev script. thanks. sorry for the confusion. will close it.